components/automate-chef-io/README.md

```diff
@@ -81,11 +88,314 @@ This will create the file and fill in the necessary frontmatter automatically.
+	| latest_report.end_time (default) | end_time |
+	| latest_report.status | status |
+	| node_name | node_name.lower |
+	*/
+	rpc ListReports(Query) returns (Reports) {
+```
+
+## Service Documentation
+
+Not in iteration #1.
+
+## Resource Documentation
+
+Given a GRPC endpoint defined like this in the proto files:
+
+```go
+>.rpc ListReports(Query) returns (Reports) {
+>.>.option (google.api.http) = {
+>.>.>.post: "/compliance/reporting/reports"
+>.>.>.body: "*"
+>.>.};
+>.};
+```
+
+* For the first iteration of the Chef Automate API docs, we need summaries and descriptions for all user-facing end points. This can be done two ways, in the comment style (preferred) or the attribute style.
+
+* Comment style of endpoint documentation supports multiline fields. Attribute style does not.
+
+* Descriptions support Github Flavored Markdown, which means that you can use lists, tables, code-blocks and more.
+
+### Comment Style Descriptions
+
+```go
+  /*
+  List reports
+  This is the description.
+  The description can span multiple lines and can contain `Github Flavored Markdown`
+  */
+  rpc ListReports(Query) returns (Reports) {
+    option (google.api.http) = {
+      post: "/compliance/reporting/reports"
+      body: "*"
+    };
+    option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {
+      tags: "Reporting"
+    };
+  };
+```
+
+### Attribute Style Descriptions
+
+```go
+rpc ListReports(Query) returns (Reports) {
+    option (google.api.http) = {
+      post: "/compliance/reporting/reports"
+      body: "*"
+    };
+    option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {
+      tags: "Reporting"
+      summary: "List reports"
+      description: "This is the description.\nThe description can span multiple lines and can contain Github Flavored Markdown."
+    };
+  };
+```
+
+Resource documentation provides RPC-level description. It is sometimes also called "method description" because it documents the what you can do with a specific API HTTP Method.
+
+Use block/leading comments for RPC-level/method documentation.
+
+Compare the leading comments to the content of the API documentation for the `ReadNode(ID)` resource.
+
+```go
+	/*
+	Fetch a node #(summary)
+
+	Fetch a specific node by id. #(description)
+	Does not support filtering, pagination or sorting.
+	*/
+	rpc ReadNode(Id) returns (Node) {
+```
+
+The rendered documentation:
+
+![](./static/images/docs/api-resource-definition.png)
+
+### Message Level Descriptions
+
+Document fields and parameters directly above the relevant code line, using `//` inline comments
+
+## Message Documentation
+
+Given messages defined like this in the proto files:
+
+```go
+message Dog {
+  // The dog's name.
+  string name = 1;
+  // Intentionally blank.
+  Food favorite_food = 2;
+  // Types of food the dog is normally fed.
+  repeated Food typical_foods = 3;
+}
+// Stats on the statuses of nodes matching the filters.
```

Comment:
```suggestion
// A type of food eaten by a Dog
```

Not super important what this content is because it is a fake example, but the current text copy-pasted from elsewhere is kind of confusing.